tests/manage/mcg/test_namespace_crd.py

```diff
@@ -982,13 +983,28 @@ def test_respin_mcg_pod_and_check_data_integrity_crd(
         original_folder = test_directory_setup.origin_dir
         result_folder = test_directory_setup.result_dir
         logger.info("Upload files to NS bucket")
-        self.write_files_to_pod_and_upload(
+        uploaded_obj_list = self.write_files_to_pod_and_upload(
             mcg_obj,
             awscli_pod_session,
             bucket_to_write=ns_bucket,
             original_dir=original_folder,
             amount=3,
         )
+        logger.info("list uploaded objects")
+        for i in range(5):
+            obj_list = list_objects_from_bucket(
+                awscli_pod_session,
+                f"s3://{ns_bucket}",
+                s3_obj=mcg_obj,
+            )
+            if uploaded_obj_list == obj_list:
+                logger.info(f"Object list {obj_list}")
+                break
+            else:
+                logger.info("Objects are not synced yet retrying after 10 secs")
+                sleep(10)
+        else:
+            assert "Failed to sync objects"
```

Comment:
This will never throw an exception, so the test won't fail here even if the upload wasn't complete. 

Python assertions throw an AssertError exception only if the expression evaluates as a "Falsey" value such as None, 0 or in this case, an empty string (check https://url.corp.redhat.com/f49ccda for info on "Falsey" and "Truthy" values in Python).

What you could do to fix this is to raise an exception instead:

```suggestion
            assert uploaded_obj_list == obj_list
            if uploaded_obj_list == obj_list:
                logger.info(f"Object list {obj_list}")
                break
            else:
                logger.info("Objects are not synced yet retrying after 10 secs")
                sleep(10)
        else:
            raise UnexpectedBehaviour("Failed to sync objects")
```

And while we're at it, adding a more descriptive Exception to ocs_ci.ocs.exceptions such as "SyncIncomplete" could be useful for future cases as well.

Reply:
Also, this kind of logic could be very useful in future test cases, so consider moving this to a new `bucket_utils.py` function. Something with a header like:

`assert_objects_in_bucket(bucket_name, objects_list, mcg_obj, s3pod,  tries=5, delay=10)`